index/scorch/segment/mem/segment.go

```diff
@@ -87,12 +87,17 @@ type Segment struct {
 	// stored field array positions
 	//  docNum -> field id -> slice of array positions (each is []uint64)
 	StoredPos []map[uint16][][]uint64
+
+	// for marking the docValue override status
+	// field id -> status
+	DocValueFields map[uint16]bool
```

Comment:
since these are just bools (not that large) and the key of the map is field id (integer, and not that many of them) could we just use a slice instead?